goal_src/jak2/engine/physics/trajectory.gc

```diff
@@ -7,3 +7,538 @@
+                              )
+                      (search-info-flag)
+                      arg3
+                      (the-as vector #f)
+                      arg4
+                      )
+                    )
+              )
+          (if (type? gp-2 process-focusable)
+              (empty)
+              )
+          )
+        )
+      )
+    )
+  (none)
+  )
+
+(defmethod combo-tracker-method-12 combo-tracker ((obj combo-tracker) (arg0 vector) (arg1 vector) (arg2 process) (arg3 time-frame))
+  (initialize obj arg0 arg1)
+  (set! (-> obj target) (process->handle arg2))
+  (set! (-> obj move-start-time) arg3)
+  obj
+  )
+
+(defmethod point-tracker-method-11 combo-tracker ((obj combo-tracker) (arg0 vector) (arg1 vector) (arg2 vector) (arg3 float))
+  (local-vars (at-0 int))
+  (with-pp
+    (rlet ((vf0 :class vf)
+           (vf1 :class vf)
+           (vf2 :class vf)
+           )
+      (init-vf0-vector)
+      (let ((v1-1 (point-tracker-method-10 obj (new 'stack-no-clear 'vector) arg1 arg2 arg3)))
+        (vector-! arg0 v1-1 arg1)
+        )
+      (let ((a1-4 (new 'stack-no-clear 'event-message-block)))
+        (set! (-> a1-4 from) (process->ppointer pp))
+        (set! (-> a1-4 num-params) 0)
+        (set! (-> a1-4 message) 'nav-control)
+        (let* ((s3-0 (send-event-function (handle->process (-> obj target)) a1-4))
+               (s4-1 (if (type? s3-0 nav-control)
+                         s3-0
+                         )
+                     )
+               )
+          (when s4-1
+            (let ((a2-3 ((method-of-type nav-control nav-control-method-11) (the-as nav-control s4-1) arg1)))
+              (if a2-3
+                  ((method-of-type nav-control nav-control-method-20) (the-as nav-control s4-1) arg1 a2-3 arg0 #x45000000 #f #f)
+                  (set! (-> arg0 quad) (the-as uint128 0))
+                  )
+              )
+            )
+          )
+        )
+      (let ((v1-12 arg0))
+        (.lvf vf1 (&-> arg0 quad))
+        (let ((f0-0 (-> pp clock frames-per-second)))
+          (.mov at-0 f0-0)
+          )
+        (.mov vf2 at-0)
+        (.mov.vf vf1 vf0 :mask #b1000)
+        (.mul.x.vf vf1 vf1 vf2 :mask #b111)
+        (.svf (&-> v1-12 quad) vf1)
+        )
+      arg0
+      )
+    )
+  )
+
+(defun traj2d-calc-initial-speed-using-tilt ((arg0 traj2d-params))
+  (let ((f0-4 (* (- (* (-> arg0 x) (tan (-> arg0 initial-tilt))) (-> arg0 y)) (/ 2.0 (-> arg0 gravity)))))
+    (cond
+      ((< 0.0 f0-4)
+       (set! (-> arg0 time) (sqrtf f0-4))
+       (set! (-> arg0 initial-speed) (/ (-> arg0 x) (* (-> arg0 time) (cos (-> arg0 initial-tilt)))))
+       #t
+       )
+      (else
+        (set! (-> arg0 time) 0.0)
+        (set! (-> arg0 initial-speed) 0.0)
+        #f
+        )
+      )
+    )
+  )
+
+(defun traj3d-calc-initial-velocity-using-tilt ((arg0 traj3d-params))
+  (vector-! (-> arg0 diff) (-> arg0 dest) (-> arg0 src))
+  (let ((s5-0 (new 'stack-no-clear 'trajectory)))
```

Comment:
I think this should be a `traj2d-params` (or the arg type of `traj2d-calc-initial-speed-using-tilt` is wrong)